code_challenges/code_wars/trigrams.py

```diff
@@ -0,0 +1,19 @@
+# https://www.codewars.com/kata/55d8dc4c8e629e55dc000068
+
+
+def extract_trigrams(words):
+    stopping_point = len(words) - 2
+    if len(words) <= 3:
+        return ""
+    spaces_replaced = ""
+    for i in words:
+        if i == " ":
+            i = "_"
+        spaces_replaced += i
```

Comment:
`i` is really only ever a good variable name if it's referring to an index, which is an integer. The same is true for `i, j and k`.  In this case actually is `i`; what would describe the value it stores?

Excellent use of the fundamentals to modify the input string!  Now read the [str.replace](https://docs.python.org/3/library/stdtypes.html#str.replace) and see if you can convert your solution to use it.  *Your solution isn't worse than the builtin string method, the time and memory cost is likely exactly the same.